src/Backend/Core/Layout/Templates/macros.html.twig

```diff
@@ -112,3 +107,17 @@
   </div>
 {% endmacro %}
 
+{% macro emptyState(addItemLink, addItemText, noItemsText, addItemLinkExtraAttributes) %}
+  {% set addItemText = addItemText|default('msg.AddItem'|trans|raw) %}
+  {% set noItemsText = noItemsText|default('msg.NoItems'|trans|raw) %}
+
+  <div class="empty-state">
+    <img class="empty-state-illustration mb-4" src="{{ BACKEND_CORE_URL }}/Layout/images/empty-state.svg" alt="">
+    <p class="text-black mb-2"><strong>{{ noItemsText|raw }}</strong></p>
+    {% if not addItemLink is null %}
+      <a class="empty-state-add" href="{{ addItemLink }}"{% for attrname, attrvalue in addItemLinkExtraAttributes %} {{ attrname }}="{{ attrvalue }}"{% endfor %}>
```

Comment:
just a thought
i like that you can add other attributes, but you can't add extra classes to your link, wouldn't anybody want to do that?

Reply:
Mm I do not thinks this will be an issue. Classes are for styling(css). We strive for equality in the code. So I suggest to won't let them add css class to maintain unity. They can add attributes, id etc when you have to link from PHP or JS